src/MongoDB/Cursor.c

```diff
@@ -312,7 +312,11 @@ PHP_METHOD(Cursor, current)
 	data = php_phongo_cursor_get_current_data(intern);
 
 	if (data) {
-		ZVAL_COPY_DEREF(return_value, data);
+		if (Z_ISUNDEF_P(data)) {
```

Comment:
At first glance, I wondered if it would be possible for this function to still leave `return_value` unset.

`php_phongo_cursor_get_current_data` always returns a non-null pointer to a block of memory within `intern`, which itself is never null (we'd hit a segfault if so). In that case, I think can should drop the outer `if (data)` for readability -- optionally leave a comment that "data points to some memory within intern and is always non-null".

Reply:
Removed the check 👍